internal/probing/probe.go

```diff
@@ -0,0 +1,133 @@
+package probing
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type Prober interface {
+	Probe(obj *unstructured.Unstructured) (success bool, message string)
+}
+
+type List []Prober
```

Comment:
This name might be a little bit confusing because of the `list.List` type. Maybe a different name would be better

Reply:
When used it's always `probing.List`, also when using the `parse` methods, nobody will directly interact with this type anyway.
Let me make them package-private... :)